tests/test_set_config.py

```diff
@@ -399,3 +399,102 @@ def test_values_needed_f90nml(): #pylint: disable=unused-variable
     salad.appetizer
 """
     assert result == outcome
+
+def test_cfg_to_yaml_conversion(): #pylint: disable=unused-variable
+    ''' Test that .a cfg file can be used to create a yaml object.'''
+    input_file = os.path.join(uwtools_file_base, pathlib.Path("fixtures/simple2_yaml.cfg"))
+
+    with tempfile.TemporaryDirectory(dir='.') as tmp_dir:
+
+        args = ['-i', input_file, '--dry_run', '--input_file_type', '.yaml']
+
+        outstring = io.StringIO()
+        with redirect_stdout(outstring):
+            set_config.create_config_obj(args)
+        result = outstring.getvalue()
+
+        expected = config.YAMLConfig(input_file)
+        expected_file = f'{tmp_dir}/test.yaml'
+        expected.dump_file(expected_file)
+
+        assert result.rstrip('\n') == str(expected)
```

Comment:
I think from our templater experience, we want to make sure our files do end with a new line. 

I think it makes sense to strip this on here to check for content independently of the new line.

Would you mind adding just one additional assert in here that checks that the expected file ends with a new line? I think that would be sufficient to check the YAML dump_file method.